lib/src/perf_benchmark_base.dart

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:io';
+
+import 'benchmark_base.dart';
+import 'score_emitter.dart';
+
+class PerfBenchmarkBase extends BenchmarkBase {
```

Comment:
And you need to write a CHANGELOG entry about this new feature and bump the minor version number.